nca/NetworkConfig/NetworkConfigQueryRunner.py

```diff
@@ -117,59 +119,115 @@ def _execute_pair_configs_query(self, query_type, config1, config2, cmd_line_fla
-        res = 0
-        output = ''
-        queries_not_executed = 0
+        res, output, queries_not_executed = self._init_query_results_values()
         for ind1 in range(len(self.configs_array) - 1):
             config1 = self.configs_array[ind1]
             for ind2 in range(ind1 + 1, len(self.configs_array)):
                 query_res, query_output, query_not_executed = self._execute_pair_configs_query(
                     self.query_name, self._get_config(config1), self._get_config(self.configs_array[ind2]),
                     cmd_line_flag)
-                res += query_res
-                output += query_output + '\n'
-                queries_not_executed += query_not_executed
-        return res, output, queries_not_executed
+                res, output, queries_not_executed = \
+                    self._update_query_results_after_one_iteration(res, query_res, output, query_output,
+                                                                   queries_not_executed, query_not_executed)
+        return self._return_final_query_results(res, output, queries_not_executed)
 
     def _run_query_on_all_pairs(self):
-        res = 0
-        output = ''
-        queries_not_executed = 0
+        res, output, queries_not_executed = self._init_query_results_values()
         for config1 in self.configs_array:
             for config2 in self.configs_array:
                 if config1 != config2:
                     query_res, query_output, query_not_executed = self._execute_pair_configs_query(
                         self.query_name, self._get_config(config1), self._get_config(config2))
-                    res += query_res
-                    output += query_output + '\n'
-                    queries_not_executed += query_not_executed
+                    res, output, queries_not_executed = \
+                        self._update_query_results_after_one_iteration(res, query_res, output, query_output,
+                                                                       queries_not_executed, query_not_executed)
+
+        return self._return_final_query_results(res, output, queries_not_executed)
+
+    # following 3 def-s are for avoiding code repetition in previous 4 def-s
+    def _init_query_results_values(self):
+        """
+        initializes the query variable results as following
+        res : 0 (the numerical result)
+        output : will be initialized to an empty list if the output format is json, otherwise, empty str
+        (the formatted explanation result of the query)
+        queries not executed : 0 (the number of not executed queries)
+        rtype: int , Union[list, str], int
+        """
+        # json results will be appended in a list, and finally be dumped into string to
+        # ensure all results are written under one top level object to get a fixed json output format
+        output = [] if self.output_configuration.outputFormat == 'json' else ''
```

Comment:
I think we should prefer holding the same types of objects.
We can hold a list of outputs and eventually use strings join by newline or json dump , depending on the output format.